### PR TITLE
Shell Hello World

### PR DESCRIPTION
**Pull request by steren**

### CODE REVIEW

**Comment by steren** [comment]
@broady 
This is what is done in the [Knative Go sample](https://github.com/knative/docs/blob/master/serving/samples/helloworld-go/helloworld.go#L29) and in the [App Engine Go hello world](https://github.com/GoogleCloudPlatform/golang-samples/blob/master/appengine/go11x/helloworld/helloworld.go#L32)

Should we change these too?
Would you mind suggesting the best replacement (I am a Go n00b)

File: serving/samples/helloworld-shell/invoke.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+)
+
+func handler(w http.ResponseWriter, r *http.Request) {
+	out, err := exec.Command("/bin/sh", "script.sh").Output()
+	if err != nil {
+		log.Fatal(err)
```
Comment (broady) [inline serving/samples/helloworld-shell/invoke.go]: This log.Fatal.

Suggestion, replace with

w.WriteHeader(500)

File: serving/samples/helloworld-shell/invoke.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	}
+	fmt.Fprintf(w, "%s\n", out)
```
Comment (broady) [inline serving/samples/helloworld-shell/invoke.go]: no need to format as a string, just

w.Write(out)

File: serving/samples/helloworld-shell/invoke.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+)
+
+func handler(w http.ResponseWriter, r *http.Request) {
+	out, err := exec.Command("/bin/sh", "script.sh").Output()
```
Comment (broady) [inline serving/samples/helloworld-shell/invoke.go]: Output() discards stderr, is that intentional?

Comment (steren) [reply serving/samples/helloworld-shell/invoke.go]: I am not sure. I assumed stderr would show up in the logs. In which case, yes I prefer for errors to not be returned to the user.

Comment (broady) [reply serving/samples/helloworld-shell/invoke.go]: Stderr from the subprocess will only appear in actual stderr if it's redirected explicitly.

cmd := exec.CommandContext(...)
cmd.Stderr = os.Stderr
...

Comment (steren) [reply serving/samples/helloworld-shell/invoke.go]: Done, I confirm that stderr lands in the error logs

File: serving/samples/helloworld-shell/README.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    import (
+        "fmt"
+        "log"
+        "net/http"
+        "os"
+        "os/exec"
+    )
+
+    func handler(w http.ResponseWriter, r *http.Request) {
+        cmd := exec.Command("/bin/sh", "script.sh")
```
Comment (broady) [inline serving/samples/helloworld-shell/README.md]:
```suggestion
cmd := exec.CommandContext(r.Context(), "/bin/sh", "script.sh")
```

File: serving/samples/helloworld-shell/invoke.go [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+)
+
+func handler(w http.ResponseWriter, r *http.Request) {
+    cmd := exec.Command("/bin/sh", "script.sh")
```
Comment (broady) [inline serving/samples/helloworld-shell/invoke.go]:
```suggestion
cmd := exec.CommandContext(r.Context(), "/bin/sh", "script.sh")
```

needs gofmt, too.

**Comment by broady** [comment]
LGTM

**Comment by steren** [comment]
CLA bot is grumpy, not sure how to fix that, I'll leave this to the repo maintainers

**Comment by broady** [comment]
> We need to confirm that all authors are ok with their commits being contributed to this project. 

confirmed

repo owner needs to add `cla: yes` to labels (you should also add `cla: no` as a possible label, but not to this issue)

File: serving/samples/helloworld-shell/README.md [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+1. After the build has completed and the container is pushed to docker hub, you
+   can deploy the app into your cluster. Ensure that the container image value
+   in `service.yaml` matches the container you built in the previous step. Apply
+   the configuration using `kubectl`:
+
+   ```shell
+   kubectl apply --filename service.yaml
+   ```
+
+1. Now that your service is created, Knative will perform the following steps:
```
Comment (carieshmarie) [inline serving/samples/helloworld-shell/README.md]:
```suggestion
1. Now that your service is created, Knative performs the following steps:
```

File: serving/samples/helloworld-shell/README.md [hunk 7]
```diff
@@ -0,0 +1,186 @@
+# Hello World - Shell sample
+
+A simple web app that executes a shell script.
+The shell script reads an env variable `TARGET` and prints `Hello ${TARGET}!`.
+If `TARGET` is not specified, it will use `World` as the `TARGET`.
```
Comment (carieshmarie) [inline serving/samples/helloworld-shell/README.md]:
```suggestion
If the `TARGET` environment variable is not specified, the script uses `World`.
```

**Comment by steren** [comment]
@carieshmarie do you know how to make CLA bot happy?

This PR seems ready to merge.

**Comment by steren** [comment]
note that it's not about the test, it;'s about the CLA bot not being happy because Chris sent a commit in this PR, and I think you need to add the cla: yes label

**Comment by samodell** [comment]
The CLA: yes label has already been added. Not sure why the CLA bot is still upset.

**Comment by steren** [comment]
anybody is able to merge?

**Pull request merged**